user_tools/src/spark_rapids_tools/tools/qualification_stats_report.py

```diff
@@ -0,0 +1,162 @@
+    logger: Logger = field(default=None, init=False)
+    unsupported_operators_df: pd.DataFrame = field(default=None, init=False)
+    stages_df: pd.DataFrame = field(default=None, init=False)
+    result_df: pd.DataFrame = field(default=None, init=False)
+    execs_df: pd.DataFrame = field(default=None, init=False)
+    output_columns: dict = field(default=None, init=False)
+    qual_output: str = field(default=None, init=True)
+    ctxt: ToolContext = field(default=None, init=True)
+
+    def __post_init__(self) -> None:
+        self.logger = ToolLogging.get_and_setup_logger('rapids.tools.qualification.stats')
+        self.output_columns = self.ctxt.get_value('local', 'output', 'files', 'statistics')
+
+    def _read_csv_files(self) -> None:
+        self.logger.info('Reading CSV files...')
+        if self.qual_output is None:
+            qual_output_dir = self.ctxt.get_rapids_output_folder()
+        else:
+            qual_output_dir = self.qual_output
+
+        unsupported_operator_report_file = self.ctxt.get_value(
+            'toolOutput', 'csv', 'unsupportedOperatorsReport', 'fileName')
+        rapids_unsupported_operators_file = FSUtil.build_path(
+            qual_output_dir, unsupported_operator_report_file)
+        self.unsupported_operators_df = pd.read_csv(rapids_unsupported_operators_file)
+
+        stages_report_file = self.ctxt.get_value('toolOutput', 'csv', 'stagesInformation',
+                                                 'fileName')
+        rapids_stages_file = FSUtil.build_path(qual_output_dir, stages_report_file)
+        self.stages_df = pd.read_csv(rapids_stages_file)
+
+        rapids_execs_file = self.ctxt.get_value('toolOutput', 'csv', 'execsInformation',
+                                                'fileName')
+        self.execs_df = pd.read_csv(FSUtil.build_path(qual_output_dir, rapids_execs_file))
+        self.logger.info('Reading CSV files completed.')
+
+    def _convert_durations(self) -> None:
+        # Convert durations from milliseconds to seconds
+        self.unsupported_operators_df[['Stage Duration', 'App Duration']] /= 1000
+        self.stages_df[['Stage Task Duration', 'Unsupported Task Duration']] /= 1000
+
+    def _preprocess_dataframes(self) -> None:
+        self.logger.info('Preprocessing dataframes...')
+
+        # Filter out 'WholeStageCodegen' operators as the child operators are already included
+        # in the other rows
+        self.execs_df = self.execs_df[
+            ~self.execs_df['Exec Name'].str.startswith('WholeStageCodegen')]
+
+        # Split 'Exec Stages' and explode the list into separate rows so that the stageID
+        # from this dataframe can be matched with the stageID of stages dataframe
+        self.execs_df['Exec Stages'] = self.execs_df['Exec Stages'].str.split(':')
+        self.execs_df = self.execs_df.explode('Exec Stages').dropna(subset=['Exec Stages'])
+        self.execs_df['Exec Stages'] = self.execs_df['Exec Stages'].astype(int)
```

Comment:
since we exploded it and changed to one stage id per row can we rename the column to be stage id?

Reply:
done